roc/logger.go

```diff
@@ -0,0 +1,21 @@
+package roc
+
+import "fmt"
+
+type logger interface {
+	Print(v ...interface{})
+}
+
+var handler LogHandler
+
+func LogSetLevel(level LogLevel) {
+	LogSetLevelImpl(level)
+}
+
+func LogSetHandler(l logger) {
+	handler = func(level LogLevel, component string, message string) {
+		l.Print(fmt.Sprintf("%s: %s", component, message))
+	}
+
+	LogSetHandlerImpl(handler)
```

Comment:
I suggest to add level to the message as well.

Also, it seems we need to guard this with a mutex. libroc provides a thread-safe interface and LogSetHandlerImpl() itself is thread-safe. However LogSetHandler() is currently not thread-safe.

Reply:
Also, let's rename these funcs to more natural "SetLogLevel" and "SetLogHandler".

Reply:
Hm, it seems that LogSetHandlerImpl generated by c-for-go already stores the function into a global variable: https://github.com/roc-project/roc-go/blob/251b4476ed4152e41f4efac0448f304c9e84f615/roc/cgo_helpers.go#L682-L690

So we don't need to duplicate this code.

However, it also seems that the generated code is not thread-safe. We should learn whether c-for-go is capable of generating thread-safe code. If it's not, we should implement thread-safety in our wrappers. The goal is to make the entire API thread-safe, just like original C API.

Reply:
#5